Set T-deck external notifications to false by default

loredous opened the pull request:
(Third time's the charm, right?)

To protect the sanity of DefCon attendees, this change sets the notification on the T-deck to be disabled by default as requested in https://github.com/meshtastic/firmware/issues/4359

thebentern commented:
I would set this line in NodeDB.cpp to false instead. That should cover the T-Watch and T-Deck. Users will be able to turn it on after the fact with just enabling ext. notifications that way as well.
![image](https://github.com/user-attachments/assets/91cede73-147b-4e58-86a0-e7505727d484)

thebentern reviewed (changes requested):
See main comment thread

loredous commented:
> I would set this line in NodeDB.cpp to false instead. That should cover the T-Watch and T-Deck. Users will be able to turn it on after the fact with just enabling ext. notifications that way as well. 

Addressed in updated commit. Moved changes to NodeDB
